src/components/alarm/AlarmController.cpp

```diff
@@ -27,30 +27,45 @@ using namespace std::chrono_literals;
 AlarmController::AlarmController(Controllers::DateTime& dateTimeController) : dateTimeController {dateTimeController} {
 }
 
-APP_TIMER_DEF(alarmAppTimer);
-
 namespace {
-  void SetOffAlarm(void* p_context) {
-    auto* controller = static_cast<Pinetime::Controllers::AlarmController*>(p_context);
-    if (controller != nullptr) {
-      controller->SetOffAlarmNow();
-    }
+  void SetOffAlarm(TimerHandle_t xTimer) {
+    auto controller = static_cast<Pinetime::Controllers::AlarmController*>(pvTimerGetTimerID(xTimer));
+    controller->SetOffAlarmNow();
   }
 }
 
 void AlarmController::Init(System::SystemTask* systemTask) {
-  app_timer_create(&alarmAppTimer, APP_TIMER_MODE_SINGLE_SHOT, SetOffAlarm);
   this->systemTask = systemTask;
+  alarmAppTimer = xTimerCreate("alarmAppTm", 1, pdFALSE, this, SetOffAlarm);
 }
 
 void AlarmController::SetAlarmTime(uint8_t alarmHr, uint8_t alarmMin) {
   hours = alarmHr;
   minutes = alarmMin;
 }
 
+void AlarmController::OnAdjustTime() {
+
+  if (state != AlarmState::Set) {
+    return;
+  }
+
+  xTimerStop(alarmAppTimer, 0);
+  auto now = dateTimeController.CurrentDateTime();
+
+  if (now > alarmTime) {
+    xTimerChangePeriod(alarmAppTimer, 1, 0);
+  } else {
+    auto mSecToAlarm = std::chrono::duration_cast<std::chrono::milliseconds>(alarmTime - now).count();
+    xTimerChangePeriod(alarmAppTimer, APP_TIMER_TICKS(mSecToAlarm), 0);
```

Comment:
Use `pdMS_TO_TICKS()` instead of `APP_TIMER_TICKS()`. With this the rest of the app_timer includes should be possible to remove across all files.

Reply:
pdMS_TO_TICKS() can't be used in that context because the 32-bit arithmetic it performs would overflow for time durations over ~69 minutes.
APP_TIMER_TICKS uses 64-bit arithmetic and doesn't have that limitation.
Perhaps we can define a custom/function macro to replace it?

Reply:
I see. Maybe we could change the mSecToAlarm to seconds, so it would only need to be multiplied by `configTICK_RATE_HZ` to get the ticks? In an alarm this should be enough precision. In MotorController `pdMS_TO_TICKS` can be used.

Reply:
That would sound acceptable to me too, we may lose second-splitting accuracy, but save some bytes in the compiled code.  The alarm can be set only with minute resolution at present.